### ISSUE TITLE
allunique(::StepRangeLen) regression

### ISSUE THREAD

**Issue by sostock** [JuliaLang/julia#36849]
Julia 1.4:
```julia
julia> allunique(range(0.0, stop=0.0, length=5))
false
```

Julia 1.5-rc2 and master:
```julia
julia> allunique(range(0.0, stop=0.0, length=5))
true
```

This was introduced by #35008. The regression would be fixed by #36836. However, that PR also introduces a minor change. I will split it into two PRs, one containing the `allunique` bugfix and another one with the minor change.

**Comment by sostock** [comment]
Since the fix for this (#36850) didn’t make it into 1.5.0, should this be reopened?

**Comment by StefanKarpinski** [comment]
No, it's fixed on master, so the issue is closed. The fix PR is marked for backport, so it will be in the next 1.5 release.
